Bug 1375093 - classify itisatracker QA sites

danielhartnell opened the pull request:
Hey April,

This PR is in response to a discussion last week about the `itisatracker` websites. More context can be found here:

https://bugzilla.mozilla.org/show_bug.cgi?id=1375093#c8

Ultimately, this just gives us an opportunity to classify these sites.

danielhartnell committed:
Bug 1375093 - classify itisatracker QA sites

floatingatoll commented:
r+ from me, fwiw.

april commented:
Simply marking it as "QA" doesn't actually do anything.

There is a mapping of sites to a deviation that is applied to them (as you changed), but it doesn't do anything until you define what the "QA" deviation actually does:

https://github.com/mozilla/http-observatory-dashboard/blob/master/httpobsdashboard/conf/deviations.json

Also, could I ask for you to use a more descriptive name than "QA"? I think what you're driving at is wanting to use "Duplicate" (since it's a redirect) for two of them, and possibly a new category that is "Requires HTTP", but I might be misreading that correctly. If you need help defining the deviations, let me know.

Thanks!

floatingatoll commented:
What would be a good label for marking sites that must violate Observatory guidance to permit "testing" of unsecured and mixed resources?

april commented:
I think "Requires HTTP" is probably fine. Note that there are two possible states here:

a) supports HTTP but not HTTPS -> this has a knock-off effect on a ton of tests and will require a lot of deviations
b) supports HTTP and HTTPS, but doesn't redirect -> this should only require 0'ing out the redirection test.  Another site we have like this is: http://mixed-content-tests-mozilla.org/

floatingatoll commented:
I classify these sites *exactly* like mixed-contest-tests, mentally, so we should probably bring that one into the exception list as well.

danielhartnell committed:
add HTTP only and package repository site deviations

danielhartnell committed:
resolved merge conflicts by including all site-deviations

danielhartnell committed:
resolved merge conflicts by including all site-deviations

danielhartnell commented:
@april,

I have updated this PR to reflect the conversation above, @floatingatoll's note regarding mixed-contest-tests and this comment:

https://bugzilla.mozilla.org/show_bug.cgi?id=1314166#c4

I have implemented the "HTTP Only" deviation which sets a score modifier of `0` for `redirection-missing` and `strict-transport-security`. What do you think about including the HSTS check? It seemed reasonable that you would not care about an HSTS header in this case.

```
  "itisatracker.com": ["HTTP Only"],
  "itisatracker.org": ["HTTP Only"],
  "trackertest.org": ["HTTP Only"],
  "mixed-content-tests-mozilla.org": ["HTTP Only"],
  "pypi.pub.build.mozilla.org": ["HTTP Only", "Package Repository"],
```

I have also created a "Package Repository" deviation. I was trying to think of a good name in response to atoll's comment [here](https://bugzilla.mozilla.org/show_bug.cgi?id=1314166#c4). This name seemed generic enough that it could apply to more sites while also remaining descriptive. I'm open to any suggestions regarding the naming or the tests included within this deviation.

april commented:
Sorry, I'll get to this soon. I've been out on leave the last few weeks.  :)

danielhartnell commented:
Hey @april,

Have you had a chance to look at this yet?

Thanks!

april commented:
Oh sorry, apparently I have not. Lemme look now.  :)

april commented on httpobsdashboard/conf/deviations.json:
```diff
@@ -180,5 +180,42 @@
       "pass": true
     }
+  },
+  "HTTP Only": {
```
I have no problem with this as a deviation, but I think it would be better to use something other than `HTTP Only`. All of these sites (except pypi for some reason) _do_ support HTTPS, they just don't forcibly redirect. Something like `No Redirection` or along those lines would be fine.

april commented on httpobsdashboard/conf/deviations.json:
```diff
@@ -180,5 +180,42 @@
+          "score_description": "Allow HTTP without an HTTPS redirect.",
+          "score_modifier": 0
+        },
+        "redirection-missing": {
```
The test name is `redirection`. You can see what you would be overwritten by going to:

https://observatory.mozilla.org/analyze.html?host=mozilla.org

And looking at the `Observatory.state.results` object. Basically this is directly overwriting the contents of that dictionary/object.

april commented on httpobsdashboard/conf/deviations.json:
```diff
@@ -180,5 +180,42 @@
+  "HTTP Only": {
+    "httpobs": {
+      "tests": {
+        "strict-transport-security": {
```
For readability, could you keep these in alphabetical order, so `redirection` and then `strict-transport-security`?

april commented on httpobsdashboard/conf/deviations.json:
```diff
@@ -180,5 +180,42 @@
+      }
+    }
+  },
+  "Package Repository": {
```
Looking at this, can we instead use `Static Community Site`, rename it to `Static Site`, and update the descriptions a bit?

april commented on httpobsdashboard/conf/deviations.json:
```diff
@@ -180,5 +180,42 @@
+          "score_description": "Repository does not contain active content and does not need CSP.",
+          "score_modifier": 0
+        },
+        "subresource-integrity": {
```
This shouldn't be necessary. If a site doesn't have any cross-origin subresources, then the Observatory won't penalize you for it. I just tested pypi and that seems to be the case.
